packages/toast/src/queue.ts

```diff
@@ -1,14 +1,30 @@
 import { AddToastOptions } from "./types";
 
+type Subscriber = () => void;
+
 export class ToastQueue {
     private static queue: AddToastOptions[] = [];
-    private static isLocked = false;
-
-    public static lock = () => (this.isLocked = true);
-    public static unlock = () => (this.isLocked = false);
-    public static push = (item: AddToastOptions) => this.queue.push(item);
+    private static subscribers: Set<Subscriber> = new Set<Subscriber>();
+    private static emitChange = () => {
+        for (const subscriber of this.subscribers) {
+            subscriber();
+        }
+    };
+    public static subscribe = (subscriber: Subscriber) => {
+        this.subscribers.add(subscriber);
+        return () => {
+            this.subscribers.delete(subscriber);
+        };
+    };
+    public static getSnapshot = () => this.queue;
+    public static enqueue = (item: AddToastOptions) => {
+        this.queue = [...this.queue, item];
+        this.emitChange();
+    };
     public static shift = () => {
-        if (this.isLocked) return undefined;
-        return this.queue.shift();
+        const toast = this.queue.at(0);
+        this.queue = [...this.queue.slice(1)];
```

Comment:
Could this be simplified with `const toast = this.queue.shift();`?

Reply:
Or do we need a completely new array for this to work?

Reply:
Yup. As in my other comment, I had to figure this out the hard way